plugins/source/oracledb/client/transformer.go

```diff
@@ -1,137 +1,82 @@
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.Bytes
-}
-
-func (*Transformer) TransformFloat8(v *schema.Float8) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.Float
-}
-
-func (*Transformer) TransformInt8(v *schema.Int8) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.Int
-}
-
-func (*Transformer) TransformTimestamptz(v *schema.Timestamptz) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.Time
-}
-
-func (*Transformer) TransformJSON(v *schema.JSON) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return string(v.Bytes)
-}
-
-func (*Transformer) TransformUUID(v *schema.UUID) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	// We need a slice instead of a fixed sized array
-	bytes := make([]byte, 16)
-	copy(bytes, v.Bytes[:])
-	return bytes
-}
-
-func (*Transformer) TransformUUIDArray(v *schema.UUIDArray) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return v.String()
-}
-
-func (*Transformer) TransformInt8Array(v *schema.Int8Array) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return v.String()
-}
-
-func (*Transformer) TransformCIDR(v *schema.CIDR) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.String()
-}
-
-func (*Transformer) TransformInet(v *schema.Inet) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.String()
-}
-
-func (*Transformer) TransformMacaddr(v *schema.Macaddr) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.String()
-}
-
-func (*Transformer) TransformText(v *schema.Text) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-	return v.Str
-}
-
-func (*Transformer) TransformCIDRArray(v *schema.CIDRArray) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return v.String()
-}
-
-func (*Transformer) TransformInetArray(v *schema.InetArray) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return v.String()
-}
-
-func (*Transformer) TransformMacaddrArray(v *schema.MacaddrArray) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return v.String()
-}
-
-func (*Transformer) TransformTextArray(v *schema.TextArray) any {
-	if v.Status != schema.Present {
-		return nil
-	}
-
-	return v.String()
+func GetValue(arr arrow.Array, i int) (any, error) {
+	if arr.IsNull(i) || !arr.IsValid(i) {
+		return nil, nil
+	}
+	switch a := arr.(type) {
+	case *array.Boolean:
+		return a.Value(i), nil
+	case *array.Int8:
+		return a.Value(i), nil
+	case *array.Int16:
+		return a.Value(i), nil
+	case *array.Int32:
+		return a.Value(i), nil
+	case *array.Int64:
+		return a.Value(i), nil
+	case *array.Uint8:
+		return a.Value(i), nil
+	case *array.Uint16:
+		return a.Value(i), nil
+	case *array.Uint32:
+		return a.Value(i), nil
+	case *array.Uint64:
+		return a.Value(i), nil
+	case *array.Float32:
+		return a.Value(i), nil
+	case *array.Float64:
+		return a.Value(i), nil
+	case *array.String:
```

Comment:
LargeString?

IMO, the better way is to check for `array.BinaryLike` and then use `IsUtf8` to verify whether it's string or byte array

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/11115/commits/e034e620676f8f91cee5e04c15241c7c11cb7bc2. Not sure I follow the `array.BinaryLike` comment

Reply:
@erezrokah 
For the `array.BinaryLike`:
```go
case array.BinaryLike:
	if a.DataType().(arrow.BinaryDataType).IsUtf8() {
		return arrow.BinaryTypes.String
	}
	return arrow.BinaryTypes.Binary
```